i3pystatus/nicehash_speed.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+import requests
+import re
+import time
+import json
+
+from i3pystatus import IntervalModule, formatp
+from i3pystatus.core.util import internet, require, user_open
+
+__author__ = "Fedor Marchenko"
+__email__ = "mfs90@mail.ru"
+__date__ = "Nov 09, 2017"
+
+
+class NiceHashSpeed(IntervalModule):
+    """Module for monitoring you miner statistic on nicehash.com
+    Using public API <https://www.nicehash.com/doc-api>.
+
+    .. rubric:: Available formatters
+
+    * {addr}
+    * {algo}
+    * {status}
+    * {speed}
+
+    """
+
+    settings = (
+        ('addr', 'You BTC address'),
+        ('algo', 'Algorithm number, for showing speed per one algorithm'),
+        ('format', 'Format string used for output'),
+        ('color', 'Standard color'),
+        ('colorize', 'Enable color change on price increase/decrease'),
+        ('color_up', 'Color for price increases'),
+        ('color_down', 'Color for price decreases'),
+        ('interval', 'Update interval.'),
+        'status'
+    )
+    addr = '17a212wdrvEXWuipCV5gcfxdALfMdhMoqh'
+    # Default: 20 = DaggerHashimoto
+    algo = 20  # List of algorithm numbers may be looking on https://www.nicehash.com/doc-api.
+    format = '{addr:.6}: {algo}[ {status}] {speed}'
+    interval = 60
+    color = '#FFFFFF'
+    colorize = False
+    color_up = '#00FF00'
+    color_down = '#FF0000'
+    status = {
+        'speed_up': '▲',
+        'speed_down': '▼',
+    }
+
+    _prev_speed = 0.0
+    _prev_status = ''
+    _prev_color = '#FFFFFF'
+
+    _api_url = 'https://api.nicehash.com/api?method=stats.provider.ex&addr={addr}&from={from_unixtime}'
+    _api_query_interval = 3
+
+    def __init__(self, *args, **kwargs):
+        super(NiceHashSpeed, self).__init__(*args, **kwargs)
+        self.on_leftclick = [
+            'open_something',
+            'https://www.nicehash.com/miner/{}'.format(self.addr)
+        ]
+
+    def fetch_data(self):
+        """Fetching statistic data from nicehash API for you BTC address
+
+        :return: Return JSON data with statistic information about miners
+        """
+        with open('/tmp/nhapi.json', 'w+') as f:
```

Comment:
Do you really need to write out the `fetch_date_time` to a file? You could just cache it in memory.